developers/weaviate/concepts/indexing/hnsw-indexes.mdx

```diff
@@ -0,0 +1,156 @@
+image: og/docs/indexing.jpg
+# tags: ['basics']['indexes']
+---
+
+import HNSWIntro from '/_includes/indexes/hnsw-intro.mdx';
+
+<HNSWIntro/>
+
+import HNSWHow from '/_includes/indexes/hnsw-how.mdx';
+
+<HNSWHow/>
+
+To configure an HNSW index, see: [Configure HNSW indexes](/developers/weaviate/configuration/indexing-vector/hnsw-indexes).
+
+## Overview
+
+HNSW is an algorithm that works on multi-layered graphs. It is also an index. An HNSW index is the vector index that Weaviate creates when it applies the HNSW algorithm to the vectors in a collection. Weaviate's [implementation](https://github.com/weaviate/weaviate/tree/main/adapters/repos/db/vector/hnsw) offers full [CRUD-support](https://db-engines.com/en/blog_post/87).
+
+When Weaviate builds the index, the HNSW algorithm creates a series of layers. The layers improve search because the database engine doesn't have to scan the entire vector space at query time. Instead, the HNSW algorithm uses the layers to build a list of approximate nearest neighbors (ANN) quickly and efficiently.
+
+### Example - HNSW search
+
+Consider this diagram of a vector search that uses HNSW.
+
+![HNSW layers](../img/hnsw-layers.svg "HNSW layers")
+
+Layer zero is the lowest layer. Layer zero contains every object in the database, and the objects are well connected to each other.
+
+Some of the objects are also represented in the layers above layer zero. Each layer above layer zero has fewer objects and fewer connections.
+
+When HNSW searches the graph, it starts at the highest layer. The algorithm finds the closest matching data points in the highest layer. Then, HNSW goes one layer deeper, and finds the closest matching data points in the lower layer that correspond to the objects in the higher layer. These are the nearest neighbors.
+
+The HNSW algorithm searches the lower layer and creates list of nearest neighbors. The nearest neighbors list is the starting point for a similar search on the next layer down. The process repeats until the search reaches the lowest (deepest) layer. Finally, the HNSW algorithm returns the data objects that are closest to the search query.
+
+Since there are relatively few data objects on the higher layers, HNSW 'jumps' over large amounts of data that it doesn't need to search. In contrast, when a data store has only one layer, the search algorithm can't skip unrelated objects. Flat hierarchies mean the search engine has to scan significantly more data objects even though those object are unlikely to match the search criteria.
+
+Weaviate's HNSW implementation is a very fast, memory efficient, approach to similarity search. The memory cache only stores the highest layer of the index instead of storing all of the data objects from the lowest layer. As a search moves from a higher layer to a lower one, HNSW only adds the data objects that are closest to the search query. This means HNSW uses a relatively small amount of memory compared to other search algorithms.
```

Comment:
> Weaviate's HNSW implementation is a very fast, memory efficient, approach to similarity search. The memory cache only stores the highest layer of the index instead of storing all of the data objects from the lowest layer. 

Is this true?